pkg/cmd/experimental/bundlesecret/bundle-secret.go

```
@@ -0,0 +1,117 @@
+			}
+			for _, isspace := range sourcelist {
+				if isspace == "" {
+					err = fmt.Errorf("No spaces allowed.  Separate source names with commas, no spaces.")
+					checkErr(err)
+				}
+			}
+			secretdata := make(map[string][]byte)
+			for _, sourceitem := range sourcelist {
+				sourceitem = strings.TrimSuffix(sourceitem, "/")
+				info, err := os.Stat(sourceitem)
+				checkErr(err)
+				if info.IsDir() {
+					filelist, err := ioutil.ReadDir(sourceitem)
+					checkErr(err)
+					for _, item := range filelist {
+						if !item.IsDir() {
+							readValidFiles(sourceitem, item.Name(), item.Size(), secretdata)
+						}
+					}
+				} else {
+					readValidFiles(sourceitem, info.Name(), info.Size(), secretdata)
+				}
+			}
+			secretobj := kapi.Secret{
+				ObjectMeta: kapi.ObjectMeta{Name: objname},
+				Data:       secretdata,
+			}
+			totalsize := 0
+			for _, bytes := range secretdata {
+				totalsize += len(bytes)
+			}
+			if totalsize == 0 {
+				err = fmt.Errorf("No data added to secret.  No files read.")
+				checkErr(err)
+			}
+			if totalsize > MaxSecretSize {
+				err = fmt.Errorf("Secret exceeds max size")
+				checkErr(err)
+			}
+			setDefaultPrinter(cmd)
+			err = f.Factory.PrintObject(cmd, &secretobj, out)
+			checkErr(err)
+		},
+	}
+	cmd.Flags().VarP(&sourcelist, "source", "f", "Comma-separated list of filenames, directories to use as source of secret data")
+	kcmdutil.AddPrinterFlags(cmd)
+	return cmd
+}
+func setDefaultPrinter(c *cobra.Command) {
+	flag := c.Flags().Lookup("output")
+	if len(flag.Value.String()) == 0 {
+		flag.Value.Set("json")
+	}
+}
+func checkErr(err error) {
+	if err == nil {
+		return
+	}
+	fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+	os.Exit(1)
+}
+
+func readValidFiles(fullpath string, filename string, filesize int64, datamap map[string][]byte) {
+	file := fmt.Sprint(fullpath, "/", filename)
+	if !util.IsDNS1123Subdomain(filename) {
+		fmt.Printf("\n%s is not a valid DNS Subdomain, skipping...\n", file)
```

if they give us an invalid file, we should probably return an error, not skip it...